File: api/src/types/sight.rs
#[derive(Debug, Clone, Copy, PartialEq, sqlx::Type, serde::Serialize, serde::Deserialize)]
#[sqlx(type_name = "sight", rename_all = "snake_case")]
#[serde(rename_all = "snake_case")]
pub enum Sight {
    Blindsight,
    Darkvision,
    Tremorsense,
    Truesight,
}

File: view/src/lib/types/Sight.ts
export enum Sight {
  Blindsight = "blindsight",
  Darkvision = "darkvision",
  Tremorsense = "tremorsense",
  Truesight = "truesight",
}

export const DisplaySight: Readonly<Record<Sight, string>> = {
  [Sight.Blindsight]: "Blindsight",
  [Sight.Darkvision]: "Darkvision",
  [Sight.Tremorsense]: "Tremorsense",
  [Sight.Truesight]: "Truesight",
} as const;

export const Sights: Sight[] = Object.values(Sight);
